keps/762-ResourceClaimTemplate/README.md

```diff
@@ -0,0 +1,405 @@
+              restartPolicy: OnFailure
+              # Example Pod-level ResourceClaimTemplate (for comparison)
+              # resourceClaimTemplates:
+              # - metadata:
+              #     name: pod-gpu-template
+              #   spec:
+              #     resourceClassName: nvidia-gpu
+          resourceClaimTemplates:  # NEW: JobSet-level ResourceClaimTemplates (added inside JobSpec)
+            - metadata:
+                name: imex-channel
+              spec:
+                resourceClassName: imex-nvlink-domain
+                containers:
+                  - worker
+            - metadata:
+                name: shared-data
+              spec:
+                resourceClassName: shared-data-resource
+```
+
+#### Behavior Changes
+1. For each `Job` created from the `replicatedJobs`, the controller creates a `ResourceClaim` per
+`resourceClaimTemplates` entry (e.g., `gpu-training-jobset-job-0-imex-channel`).
+
+2. An `ownerReference` is added from the `ResourceClaim` to the `Job`.
+
+3. The controller modifies the Job's Pod spec. For the imex-channel claim,
+only the worker container gets a reference. For the shared-data claim, all containers get a reference.
+
+4. `ResourceClaim`s are deleted when the `Job` completes (default policy).
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### IMEX User Story
+
+As an engineer running simulations that require an IMEX channel within an NVLink domain,
+I want the JobSet to create a ResourceClaim representing that channel, and have all Pods
+within the Job configured to use that channel.
+
+#### TPU Slice User Story
+
+As a researcher running distributed training on a multi-node TPU slice, I want
+to define a JobSet that automatically provisions a TPU slice ResourceClaim
+shared by all Pods of a Job, ensuring my training job has the necessary multi-node resources.
+
+#### Kubeflow User Story
+
+As a Kubeflow user leveraging JobSet through common interfaces like the [Training Operator](https://github.com/kubeflow/trainer?tab=readme-ov-file#kubeflow-training-operator-v1),
+I should be able to request per-job shared resources in addition to my per-pod resources in a declarative manner,
+so that the platform can manage resources for complex multi-node, multi-pod jobs without manual intervention.
+
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+* The containers field within `resourceClaimTemplates` is optional. If omitted,
+the `ResourceClaim` will be made available to all containers in the Pod.
```

Comment:
Hmm. Probably this should be that it does not add it to any containers. Pod-level claims are valid for NICs, for example. I don't know though of multi-node pod-level claims. But still, I think less magic is better.